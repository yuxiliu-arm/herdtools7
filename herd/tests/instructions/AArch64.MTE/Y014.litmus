AArch64 Y014
(* IRG GCR_EL1 RRND=0, RGSR_EL1 tag and seed update *)
{
  int x=100;
  0:X0=x;
  0:GCR_EL1=0;
  0:RGSR_EL1=0x001400; (* SEED=0x0014 TAG=0 *)
}

P0        ;
IRG X1,X0 ;
IRG X1,X0 ;

(* Next seed calculation: *)
(* SEED before | lfsr <5:0>      | top | SEED after *)
(* 0x0014      | 0 _1 0 1 _0 0   | 1   | 0x800A     *)
(* 0x800A      | 0 _0 1 0 _1 0   | 1   | 0xC005     *)
(* 0xC005      | 0 _0 0 1 _0 1   | 0   | 0x6003     *)
(* 0x6003      | 0 _0 0 0 _1 0   | 0   | 0x3001     *)

(* After IRG, RGSR_EL1 TAG=0b0011, SEED=0x3001 *)

(* Next seed calculation (if sequential) *)
(* SEED before | lfsr <5:0>      | top | SEED after *)
(* 0x3001      | 0 _0 0 0 _0 1   | 1   | 0x9800     *)
(* 0x9800      | 0 _0 0 0 _0 0   | 0   | 0x4C00     *)
(* 0x4C00      | 0 _0 0 0 _0 0   | 0   | 0x2600     *)
(* 0x2600      | 0 _0 0 0 _0 0   | 0   | 0x1300     *)


forall (0:X1=x:t4 /\ 0:RGSR_EL1=0x130004)

