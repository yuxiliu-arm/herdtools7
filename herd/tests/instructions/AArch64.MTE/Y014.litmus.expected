Test Y014 Required
States 1
0:X1=x:t4; 0:RGSR_EL1=1245188;
Ok
Witnesses
Positive: 1 Negative: 0
Condition forall (0:X1=x:t4 /\ 0:RGSR_EL1=1245188)
Observation Y014 Always 1 0
Hash=85eeb6a1b26e2f1197afe4aec7b3f34c

